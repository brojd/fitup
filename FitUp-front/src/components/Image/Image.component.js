import React, { PropTypes } from 'react';

const Image = (props) => ((
  <img className={props.className}
       src={props.src}
       alt={props.alt}
       height={props.height}
       width={props.width}
       onClick = {props.onClick} />
));

Image.propTypes = {
  src: PropTypes.string,
  alt: PropTypes.string,
  height: PropTypes.number,
  width: PropTypes.number,
  className: PropTypes.string,
  onClick: PropTypes.func
};

Image.defaultProps = {
  height: 150,
  width: 150
};

export default Image;
