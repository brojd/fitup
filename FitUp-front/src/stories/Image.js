import React from 'react';
import { storiesOf} from '@kadira/storybook';
import Image from '../components/Image/Image.component';

storiesOf('Image', module)
  .add('logo', () => (
    <Image src='https://s22.postimg.org/hx1k588kx/Logomakr_4x27d_Q.png' alt='logo' height={58} width={80} />
  ))
  .add('slide image 1', () => (
    <Image src='http://cdn.wallpapersafari.com/42/73/bc4rDB.jpg' alt='slide_image_1' height={600} width={950} />
  ))
  .add('slide image 2', () => (
    <Image src='http://womanhealthsolutions.com/wp-content/uploads/2016/05/fitness-014.jpg' alt='slide_image_2'
           height={600} width={950} />
  ))
  .add('slide image 3', () => (
    <Image src='http://imgview.info/download/20150902/sports-fitness-model-woman-ball-1920x1080.jpg' alt='slide_image_3'
           height={600} width={950} />
  ))
  .add('slide image 4', () => (
    <Image src='http://www.advancefitness.net/wp-content/uploads/2015/11/cover3.jpg' alt='slide_image_4'
           height={600} width={950} />
  ))
  .add('gyms list element', () => (
    <Image src='https://www.thegymgroup.com/fileadmin/uploads/gym/Photos/Gym_Leicester/Leicester14.jpg'
           alt='gym picture' height={200} width={200} className='uk-thumbnail uk-border-rounded' />
  ))
  .add('classes list element', () => (
    <Image src='http://przejrzystapolska.pl/wp-content/uploads/2015/07/zumba-3.jpg'
           alt='gym picture' height={200} width={200} className='uk-thumbnail uk-border-rounded' />
  ))
  .add('trainers list element', () => (
    <Image src='http://trener-personalny.fitness/images/lukasz-gliszczynski-trener-personalny-fitness-51.jpg'
           alt='gym picture' height={200} width={200} className='uk-thumbnail uk-border-rounded' />
  ));
