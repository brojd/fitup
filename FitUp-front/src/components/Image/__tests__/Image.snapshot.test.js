import React from 'react';
import Image from '../Image.component';
import renderer from 'react-test-renderer';

test('Image renders correctly with given props', () => {
  const someFunc = jest.fn();
  const component = renderer.create(
    <Image  src='https://cdn.pixabay.com/photo/2016/10/20/17/41/travel-1756150_960_720.jpg'
            className='Image'
            onClick={someFunc}
            height={200}
            width={300}
            alt='travel_image' />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
