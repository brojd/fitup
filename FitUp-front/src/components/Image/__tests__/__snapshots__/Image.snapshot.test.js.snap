exports[`test Image renders correctly with given props 1`] = `
<img
  alt="travel_image"
  className="Image"
  height={200}
  onClick={[Function]}
  src="https://cdn.pixabay.com/photo/2016/10/20/17/41/travel-1756150_960_720.jpg"
  width={300} />
`;
